qiskit_optimization/applications/bin_packing.py

```diff
@@ -0,0 +1,93 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the bin packing."""
+
+from typing import List, Union, Optional
+
+import numpy as np
+from docplex.mp.model import Model
+
+from qiskit_optimization.algorithms import OptimizationResult
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+from qiskit_optimization.translators import from_docplex_mp
+from .optimization_application import OptimizationApplication
+
+
+class BinPacking(OptimizationApplication):
+    """Optimization application for the "bin packing" [1] problem.
+
+    References:
+        [1]: "Bin packing",
+        https://en.wikipedia.org/wiki/Bin_packing_problem
+    """
+
+    def __init__(
+        self, weights: List[int], max_weight: int, max_number_of_bins: Optional[int] = None
+    ) -> None:
+        """
+        Args:
+            weights: A list of the weights of items
+            max_weight: The maximum bin weight capacity
+            max_number_of_bins: The maximum number of bins by default equal to the number of items
+        """
+        self._weights = weights
+        self._max_weight = max_weight
+        if max_number_of_bins is None:
+            self._max_number_of_bins = len(weights)
+        else:
+            self._max_number_of_bins = max_number_of_bins
+
+    def to_quadratic_program(self) -> QuadraticProgram:
+        """Convert a bin packing problem instance into a
+        :class:`~qiskit_optimization.problems.QuadraticProgram`
+
+        Returns:
+            The :class:`~qiskit_optimization.problems.QuadraticProgram` created
+            from the bin packing problem instance.
+        """
+        mdl = Model(name="BinPacking")
+        num_bins = self._max_number_of_bins
+        num_items = len(self._weights)
+        x = mdl.binary_var_list([f"x{i}" for i in range(num_bins)])
```

Comment:
Could you revise the model to be consistent with the model of Wikipedia, which you mentioned as a reference?